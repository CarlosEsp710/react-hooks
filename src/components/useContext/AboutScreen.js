import React, { useContext } from "react";

import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleUser = () => setUser({});

  return (
    <>
      <h1>About Screen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
      <button className="btn btn-warning mt-1" onClick={handleUser}>
        Logout
      </button>
    </>
  );
};
