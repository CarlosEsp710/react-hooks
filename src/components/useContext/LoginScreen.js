import React, { useContext } from "react";

import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleUser = () =>
    setUser({
      id: 123,
      name: "Carlos Espejel",
      email: "carlosespejel71@gmail.com",
    });

  return (
    <>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleUser}>
        Login
      </button>
    </>
  );
};
