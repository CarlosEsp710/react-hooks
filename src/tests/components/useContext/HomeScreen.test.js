import React from "react";
import { mount } from "enzyme";

import { HomeScreen } from "../../../components/useContext/HomeScreen";
import { UserContext } from "../../../components/useContext/UserContext";

describe("test <HomeScreen />", () => {
  const user = {
    name: "John",
    email: "john@example.com",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test("should return component <HomeScreen />", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
