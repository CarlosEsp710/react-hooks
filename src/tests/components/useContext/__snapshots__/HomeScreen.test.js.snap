// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <HomeScreen /> should return component <HomeScreen /> 1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "John",
   "email": "john@example.com"
}
  </pre>,
]
`;
